presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxConnector.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.jmx;
+
+import com.facebook.presto.spi.NodeManager;
+import com.facebook.presto.spi.connector.Connector;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.transaction.IsolationLevel;
+import io.airlift.units.Duration;
+
+import javax.management.MBeanServer;
+
+import java.util.Set;
+
+import static com.facebook.presto.spi.transaction.IsolationLevel.READ_COMMITTED;
+import static com.facebook.presto.spi.transaction.IsolationLevel.checkConnectorSupports;
+
+public class JmxConnector
+        implements Connector
+{
+    private final JmxMetadata jmxMetadata;
+    private final JmxHistoryDumper jmxHistoryDumper;
+    private final JmxSplitManager jmxSplitManager;
+    private final JmxRecordSetProvider jmxRecordSetProvider;
+
+    public JmxConnector(
+            String connectorId,
```

Comment:
Personally I think Guice is in most times an overkill, but ok I have used it here.

One problem is that it required me to copy util class `RebindSafeMBeanServer`  from presto-raptor, please check if it's ok for you.